### ISSUE TITLE
Change Question state (student)

### ISSUE THREAD

**Issue by eLearning-Group** [studentquiz/moodle-mod_studentquiz#463]
The description says that the question cannot be changed by the author once the status is set to disapproved.
However, the state can be set to "changed" by the author and then the question can be changed again. This seems to me an error. 

If the question is disapproved it should not be possible to change the state anymore by the author.

**Comment by Frank-Koch** [comment]
The behaviour is intended. If the teacher sets the status to "approved" or "disapproved" the student cannot alter the quetion. However, if the teacher sets it to "changed" she grants the student the right again to change the question. This might be needed in the course of a discussion betreen teacher and student. After the question got changed, the teacher is expected to review it again and to grant an "approved" or "disapproved".

**Comment by eLearning-Group** [comment]
Thanks Frank for your answer. I see the intention. But the point is, if the teacher disapprove the question, the student **can change** the status to "changed" himself and then he can change the question again. 
This option should not be possible. The student should not be able to change the status.

**Comment by Frank-Koch** [comment]
I cannot confirm the described behaviour. A student CANNOT change the state of her question, only the teacher can (of course the manager as well). So I guess your student has teachers rights in this course.

**Comment by eLearning-Group** [comment]
![1  Question from Student](https://github.com/studentquiz/moodle-mod_studentquiz/assets/83763638/c1393f54-bcee-4a3a-be48-54ff3d620131)
![2  Disapproved Question View Teacher](https://github.com/studentquiz/moodle-mod_studentquiz/assets/83763638/959ab58e-2cf2-488f-a0e3-c65e3463e982)
![3  Disapproved Question View Student](https://github.com/studentquiz/moodle-mod_studentquiz/assets/83763638/a88d1a4c-738c-411d-90cc-1d1472a5795f)
![4  Student starts Quiz and sees the Question View Student](https://github.com/studentquiz/moodle-mod_studentquiz/assets/83763638/e885acd3-2bef-4e0f-80d0-ccdec8728863)
![5  Student CAN change the state](https://github.com/studentquiz/moodle-mod_studentquiz/assets/83763638/b01d4248-211f-4809-b5a0-16c9d264d607)
![6  State changed from Student](https://github.com/studentquiz/moodle-mod_studentquiz/assets/83763638/70fe8e96-096e-4244-91b2-b67ab9e71313)
![Student and Teacher Rights in the course](https://github.com/studentquiz/moodle-mod_studentquiz/assets/83763638/7d669791-597e-4860-97c2-0708426572e7)

My student has no teachers right, please check the screenshots.

**Comment by Frank-Koch** [comment]
You're right, the student can change the question in the attempt-mode. That's a bug that has to be fixed. 
The student cannot change the state in the view-mode, which I looked at first.
Thank you for pointing out.
